src/pages/home/report/ContextMenu/BaseReportActionContextMenu.js

```diff
@@ -11,38 +12,44 @@ import withLocalize, {withLocalizePropTypes} from '../../../../components/withLo
 import ContextMenuActions from './ContextMenuActions';
 
 const propTypes = {
+    type: PropTypes.string,
     ...GenericReportActionContextMenuPropTypes,
     ...withLocalizePropTypes,
 };
-
 class BaseReportActionContextMenu extends React.Component {
     constructor(props) {
         super(props);
-
         this.wrapperStyle = getReportActionContextMenuStyles(this.props.isMini);
     }
 
     render() {
         return this.props.isVisible && (
             <View style={this.wrapperStyle}>
-                {_.map(ContextMenuActions, contextAction => contextAction.shouldShow(this.props.reportAction) && (
-                    <ContextMenuItem
-                        icon={contextAction.icon}
-                        text={this.props.translate(contextAction.textTranslateKey)}
-                        successIcon={contextAction.successIcon}
-                        successText={contextAction.successTextTranslateKey
-                            ? this.props.translate(contextAction.successTextTranslateKey)
-                            : undefined}
-                        isMini={this.props.isMini}
-                        key={contextAction.textTranslateKey}
-                        onPress={() => contextAction.onPress(!this.props.isMini, {
-                            reportAction: this.props.reportAction,
-                            reportID: this.props.reportID,
-                            draftMessage: this.props.draftMessage,
-                            selection: this.props.selection,
-                        })}
-                    />
-                ))}
+                {_.map(
+                    ContextMenuActions, (contextAction) => {
+                        if (!contextAction.shouldShow(this.props.type, this.props.reportAction)) {
+                            return;
```

Comment:
NAB:  Does not look intuitive. Maybe _filtering before mapping is easier to read.